integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.spec.ts

```diff
@@ -4,4 +4,693 @@
+        },
+      ];
+      mockCartHandlerService.getSupportedDeliveryModes.and.returnValue(
+        of(mockDeliveryModes)
+      );
+
+      service.getShippingOptionParameters().subscribe((shippingOptions) => {
+        expect(shippingOptions).toBeDefined();
+        expect(shippingOptions?.shippingOptions.length).toBe(
+          mockDeliveryModes.length
+        );
+        expect(shippingOptions?.defaultSelectedOptionId).toBe(
+          mockDeliveryModes[0].code
+        );
+
+        mockDeliveryModes.forEach((mode, index) => {
+          const shippingOption = shippingOptions?.shippingOptions[index];
+          expect(shippingOption?.id).toEqual(mode.code);
+          expect(shippingOption?.label).toEqual(mode.name);
+          expect(shippingOption?.description).toEqual(mode.description);
+        });
+
+        done();
+      });
+    });
+
+    it('should handle cases with no delivery modes available', (done) => {
+      mockCartHandlerService.getSupportedDeliveryModes.and.returnValue(of([]));
+
+      service.getShippingOptionParameters().subscribe((shippingOptions) => {
+        expect(shippingOptions).toBeDefined();
+        expect(shippingOptions?.shippingOptions).toEqual([]);
+        expect(shippingOptions?.defaultSelectedOptionId).toBeUndefined();
+        done();
+      });
+    });
+  });
+
+  describe('loadProviderResources', () => {
+    it('should load the Google Pay JS API', async () => {
+      mockResourceLoaderService.loadProviderResources.and.returnValue(
+        Promise.resolve()
+      );
+
+      await service.loadProviderResources();
+
+      expect(
+        mockResourceLoaderService.loadProviderResources
+      ).toHaveBeenCalled();
+    });
+
+    it('should handle errors when loading the Google Pay JS API', async () => {
+      mockResourceLoaderService.loadProviderResources.and.returnValue(
+        Promise.reject(new Error('Load failed'))
+      );
+
+      await expectAsync(service.loadProviderResources()).toBeRejectedWithError(
+        'Load failed'
+      );
+    });
+  });
+
+  describe('isReadyToPay', () => {
+    it('should return info about readiness to pay from the Google Pay API', async () => {
+      const activeConfiguration = {};
+
+      service.initClient(activeConfiguration);
+
+      await expectAsync(service.isReadyToPay()).toBeResolvedTo({
+        result: true,
+      });
+    });
+
+    it('should handle errors from the Google Pay API', async () => {
+      spyOn(service, 'isReadyToPay').and.returnValue(
+        Promise.reject(new Error('API error'))
+      );
+
+      await expectAsync(service.isReadyToPay()).toBeRejectedWithError(
+        'API error'
+      );
+    });
+  });
+
+  describe('initClient', () => {
+    it('should initialize the Google Payment client with configurations', () => {
+      const activeConfiguration = {};
+      service.initClient(activeConfiguration);
+
+      const client = (service as any).googlePaymentClient;
```

Comment:
Do we need to cast a serivce as `any` each time?

Reply:
the thing is that it is protected property of the service, changed it to bracket notation instead of casting to any